x/ibc/04-channel/keeper/handshake_test.go

```diff
@@ -4,624 +4,634 @@ import (
-			)
+			_, _, connA, connB, _, _ = suite.coordinator.Setup(suite.chainA, suite.chainB)
+			channelA := connA.Channels[0]
+			channelCap = suite.chainA.GetChannelCapability(channelA.PortID, channelA.ID)
+
+			// set the channel's connection hops to wrong connection ID
+			channel := suite.chainA.GetChannel(channelA)
+			channel.ConnectionHops[0] = "doesnotexist"
+			suite.chainA.App.IBCKeeper.ChannelKeeper.SetChannel(suite.chainA.GetContext(), channelA.PortID, channelA.ID, channel)
 		}, false},
 		{"connection is not OPEN", func() {
-			suite.chainA.createConnection(
-				testConnectionIDA, testConnectionIDB, testClientIDA, testClientIDB,
-				connection.TRYOPEN,
-			)
-			suite.chainA.createChannel(
-				testPort1, testChannel1, testPort2, testChannel2, types.UNINITIALIZED,
-				types.ORDERED, testConnectionIDA,
-			)
+			clientA, clientB := suite.coordinator.SetupClients(suite.chainA, suite.chainB, clientexported.Tendermint)
+
+			var err error
+			connA, connB, err = suite.coordinator.ConnOpenInit(suite.chainA, suite.chainB, clientA, clientB)
+			suite.Require().NoError(err)
+
+			// create channel in init
+			channelA, _, err := suite.coordinator.ChanOpenInit(suite.chainA, suite.chainB, connA, connB, types.ORDERED)
+
+			// ensure channel capability check passes
+			suite.chainA.CreateChannelCapability(channelA.PortID, channelA.ID)
+			channelCap = suite.chainA.GetChannelCapability(channelA.PortID, channelA.ID)
 		}, false},
 		{"channel capability not found", func() {
-			suite.chainB.CreateClient(suite.chainA)
-			suite.chainA.createConnection(
-				testConnectionIDA, testConnectionIDB, testClientIDA, testClientIDB,
-				connection.OPEN,
-			)
-			suite.chainA.createChannel(
-				testPort1, testChannel1, testPort2, testChannel2, types.OPEN,
-				types.ORDERED, testConnectionIDA,
-			)
+			_, _, connA, connB, _, _ = suite.coordinator.Setup(suite.chainA, suite.chainB)
 			channelCap = capabilitytypes.NewCapability(3)
 		}, false},
 	}
 
-	for i, tc := range testCases {
+	for _, tc := range testCases {
 		tc := tc
-		i := i
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
 			suite.SetupTest() // reset
 
-			var err error
-			channelCap, err = suite.chainA.App.ScopedIBCKeeper.NewCapability(suite.chainA.GetContext(), host.ChannelCapabilityPath(testPort1, testChannel1))
-			suite.Require().NoError(err, "could not create capability")
-
 			tc.malleate()
-			err = suite.chainA.App.IBCKeeper.ChannelKeeper.ChanCloseInit(
-				suite.chainA.GetContext(), testPort1, testChannel1, channelCap,
+
+			channelA := connA.FirstOrNextTestChannel()
+
+			err := suite.chainA.App.IBCKeeper.ChannelKeeper.ChanCloseInit(
+				suite.chainA.GetContext(), channelA.PortID, channelA.ID, channelCap,
 			)
 
 			if tc.expPass {
-				suite.Require().NoError(err, "valid test case %d failed: %s", i, tc.msg)
+				suite.Require().NoError(err)
 			} else {
-				suite.Require().Error(err, "invalid test case %d passed: %s", i, tc.msg)
+				suite.Require().Error(err)
```

Comment:
ditto for all the other tests and for `NoError` too
```suggestion
				suite.Require().Error(err, tc.msg)
```

Reply:
I left this out because there is the suite.Run which includes the msg name so this information felt redundant. I can update it though

Reply:
ahh yeah, feel free to omit it then

Reply:
resolving because I'm gonna omit :+1: